OpenRA.Mods.Cnc/Traits/KilledByEnergyWall.cs

```diff
@@ -0,0 +1,33 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2017 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Linq;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Cnc.Traits
+{
+	[Desc("Used to connect aircraft with `EnergyWall` logic.")]
+	public class KilledByEnergyWallInfo : TraitInfo<KilledByEnergyWall> { }
+
+	public class KilledByEnergyWall : ITick
+	{
+		void ITick.Tick(Actor self)
+		{
+			var actors = self.World.ActorMap.GetActorsAt(self.Location).Where(a => a != self);
+			foreach (var actor in actors)
+			{
+				var energyWall = actor.TraitOrDefault<EnergyWall>();
```

Comment:
This kills aircraft that fly over laser fences or disabled firestorm segments!

Reply:
After thinking about this a bit more I think the best solution will be to rename this trait to `KilledByActorBelow` and add a new conditional `KillsActorsAbove` trait that it checks for.  You can then use `EnergyWall`'s `ActiveCondition` condition to enable `KillsActorsAbove` only when the firestorm is active.

Reply:
Fixed.